diesel/src/sqlite/connection/mod.rs

```diff
@@ -370,4 +385,118 @@ mod tests {
             .get_result::<(i32, i32, i32)>(&connection);
         assert_eq!(Ok((2, 3, 4)), added);
     }
+
+    use crate::sqlite::SqliteAggregateFunction;
+    use std::ops::AddAssign;
+    sql_function! {
+        #[aggregate]
+        fn my_sum(expr: Integer) -> Integer;
+    }
+
+    #[derive(Default)]
+    struct MySum<T> {
+        sum: T,
+    }
+    impl<T: Default + Copy + AddAssign> SqliteAggregateFunction<T> for MySum<T> {
+        type Output = T;
+
+        fn step(&mut self, expr: T) {
+            self.sum += expr;
+        }
+
+        fn finalize(self) -> Self::Output {
+            self.sum
+        }
+    }
+    table! {
+        my_sum_example {
+            id -> Integer,
+            value -> Integer,
+        }
+    }
+
+    #[test]
+    fn register_aggregate_function() {
+        use self::my_sum_example::dsl::*;
+
+        let connection = SqliteConnection::establish(":memory:").unwrap();
+        connection
+            .execute(
+                "CREATE TABLE my_sum_example (id integer primary key autoincrement, value integer)",
+            )
+            .unwrap();
+        connection
+            .execute("INSERT INTO my_sum_example (value) VALUES (1), (2), (3)")
+            .unwrap();
+
+        my_sum::register_impl::<MySum<i32>, _>(&connection).unwrap();
+
+        let result = my_sum_example
+            .select(my_sum(value))
+            .get_result::<i32>(&connection);
+        assert_eq!(Ok(6), result);
+    }
+
+    sql_function! { #[aggregate] fn range_max(expr1: Integer, expr2: Integer, expr3: Integer) -> Integer; }
+
+    #[derive(Default)]
+    struct RangeMax<T> {
+        max_value: Option<T>,
+    }
+    impl<T: Default + Ord + Copy + Clone> SqliteAggregateFunction<(T, T, T)> for RangeMax<T> {
+        type Output = T;
+
+        fn step(&mut self, (x0, x1, x2): (T, T, T)) {
+            let max = if x0 >= x1 && x0 >= x2 {
+                x0
+            } else if x1 >= x0 && x1 >= x2 {
+                x1
+            } else {
+                x2
+            };
+
+            self.max_value = match self.max_value {
+                Some(current_max_value) if max > current_max_value => Some(max),
+                None => Some(max),
+                _ => self.max_value,
+            };
+        }
+
+        fn finalize(self) -> Self::Output {
+            self.max_value.unwrap()
+        }
+    }
+    table! {
+        range_max_example {
+            id -> Integer,
+            value1 -> Integer,
+            value2 -> Integer,
+            value3 -> Integer,
+        }
+    }
```

Comment:
Can you add some a single new line between each of the blocks, so that for example `MySum` and the corresponding `SqliteAggregateFunction` impl are seperate by an empty line?
(This applies to the whole block)